operator/controllers/keptntask/function_utils.go

```
@@ -0,0 +1,138 @@
+					LocalObjectReference: corev1.LocalObjectReference{Name: params.SecureParameters},
+					Key:                  "SECURE_DATA",
+				},
+			},
+		})
+	}
+
+	if params.ConfigMap != "" {
+		envVars = append(envVars, corev1.EnvVar{Name: "SCRIPT", Value: "/var/data/function.ts"})
+
+		job.Spec.Template.Spec.Volumes = []corev1.Volume{
+			{
+				Name: "function-mount",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: params.ConfigMap,
+						},
+					},
+				},
+			},
+		}
+		container.VolumeMounts = []corev1.VolumeMount{
+			{
+				Name:      "function-mount",
+				ReadOnly:  true,
+				MountPath: "/var/data/function.ts",
+				SubPath:   "code",
+			},
+		}
+	} else {
+		envVars = append(envVars, corev1.EnvVar{Name: "SCRIPT", Value: params.URL})
+	}
+
+	container.Env = envVars
+	job.Spec.Template.Spec.Containers = []corev1.Container{
+		container,
+	}
+	return job, nil
+}
+
+func parseFunctionTaskDefinition(definition *klcv1alpha1.KeptnTaskDefinition) (FunctionExecutionParams, bool, error) {
+	params := FunctionExecutionParams{}
+
+	// Firstly check if this task definition has a parent object
+	hasParent := false
+	if definition.Spec.Function.FunctionReference != (klcv1alpha1.FunctionReference{}) {
+		hasParent = true
+	}
+
+	// Check if there is a ConfigMap with the function for this object
+	if definition.Status.Function.ConfigMap != "" {
+		params.ConfigMap = definition.Status.Function.ConfigMap
+	} else {
+		// If not, check if it has an HTTP reference. If this is also not the case and the object has no parent, something is wrong
+		if definition.Spec.Function.HttpReference.Url == "" && !hasParent {
+			return params, false, fmt.Errorf("No ConfigMap specified or HTTP source specified in TaskDefinition) / Namespace: %s, Name: %s ", definition.Namespace, definition.Name)
+		}
+		params.URL = definition.Spec.Function.HttpReference.Url
+	}
+
```

should we throw an error if we have both URL and CM? 
I think having CM as a winner without error is the best option tho

---

added